Domain layer

matko-flabbergast opened the pull request.

darios3 commented on app/src/main/java/com/example/reduxtestapp/data/repository/todo/TodoRepositoryImpl.kt:
```diff
@@ -39,38 +42,38 @@ data class TodoRepositoryImpl (
                 preferences[TODO_PREFERENCES_KEY] ?: "[]"
             )
-        }.first()
+        }.first().asDomain()
```
be careful with `.first()` if the collection is empty, it'll throw an exception

darios3 commented on app/src/main/java/com/example/reduxtestapp/di/Modules.kt:
```diff
@@ -2,76 +2,9 @@ package com.example.reduxtestapp.di
-
     single<Logger> {
         DefaultLogger
     }
```
if this is only DI for `common` stuff now, move it to common and name it `commonModules`

darios3 commented on app/src/main/java/com/example/reduxtestapp/domain/model/country/CountryModel.kt:
```diff
@@ -0,0 +1,49 @@
+    translations = translations.mapValues { it.value.asDomain() }
+)
+
+fun List<CountryDto>.asDomain() = map { it.asDomain() }
```
domain doesn't know about data models so move all data -> domain mappers to `data`

darios3 commented on app/src/main/java/com/example/reduxtestapp/domain/model/todo/TodoModel.kt:
```diff
@@ -0,0 +1,22 @@
+    isCompleted = isCompleted
+)
+
+fun List<TodoDto>.asDomain() = map { it.asDomain() }
```
same thing about data models

matko-flabbergast committed:
Added domain layer to app

# Conflicts:
#	app/src/main/java/com/example/reduxtestapp/data/repository/country/CountryRepositoryImpl.kt
#	app/src/main/java/com/example/reduxtestapp/redux/Actions.kt
#	app/src/main/java/com/example/reduxtestapp/redux/di/ReduxModule.kt
#	app/src/main/java/com/example/reduxtestapp/redux/middleware/CountryMiddleware.kt
